modules/vaos/app/controllers/vaos/v2/appointments_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'common/exceptions'
+
+module VAOS
+  module V2
+    class AppointmentsController < VAOS::V0::BaseController
+      before_action :validate_params, only: :index
+
+      def index
+        render json: VAOS::V2::AppointmentsSerializer.new(appointments[:data], meta: appointments[:meta])
+      end
+
+      private
+
+      def appointments_service
+        VAOS::V2::AppointmentsService.new(current_user)
+      end
+
+      def appointments
+        @appointments ||=
+          appointments_service.get_appointments(start_date, end_date, pagination_params)
+      end
+
+      def validate_params
```

Comment:
Normally require and permit are used for parameter presence validation. Any reason that was used?

Reply:
I was wondering that myself, it was pulled from the original V0 controller but can probably be changed.  @saneshark ?

Reply:
Going to push this for now, because it's blocking other work.  Will revisit ASAP.

Reply:
I think it might be worth re-evaluating the logic here. Not super critical, but nice to have.

only downside is that we don't surface all the errors and instead just the first one we ran into.